Azure - FAQ doc updated (#4123)
SumoLogic/sumologic-documentation: docs/send-data/collect-from-other-data-sources/azure-blob-storage/troubleshoot-azure-blob-storage-log-collection.md

## Error while deploying the ARM template

If you received the following error while deploying the ARM template: **The Resource 'X' under resource group 'Y' was not found.**
<br/><img src={useBaseUrl('/img/send-data/Azure_ARM-template-error.png')}  style={{border: '1px solid gray'}} alt="Azure_ARM-template-error" width="800"/>

If you attempted to redeploy the template after getting this error and used the **same storage account + resource group values** in the first failed deployment, follow these steps.

> Refer to [this video](https://sumologic.link/6e020df8d1d7), then redeploy the template, in the following way: 

Reviewer: this link doesn't work for me

Author: @kimsauce This is already updated in the [PR](https://github.com/SumoLogic/sumologic-documentation/pull/3979). PR #3979 will be merged first and this section will be updated accordingly once we Update this branch.

1. Go to the resource group where the deployment failed
1. Click on Failed deployment, and then click on Microsoft.Template(last deployment) in the next window.
1. Click **Redeploy**, provide the required fields, and click **Purchase**. You should see the successful deployment notification.<br/><img src={useBaseUrl('/img/send-data/Azure_ARM-template-troubleshoot.png')}  style={{border: '1px solid gray'}} alt="Azure_ARM-template-troubleshoot" width="800"/>

> If you get namespace invalid error make sure it follows the naming convention specified in this [doc](https://docs.microsoft.com/en-us/rest/api/servicebus/create-namespace).

Reviewer: ```suggestion
If you get namespace invalid error, make sure it follows the naming convention specified in this [doc](https://docs.microsoft.com/en-us/rest/api/servicebus/create-namespace).
```

 ]
```

> For common deployment errors, refer to the [Troubleshoot common Azure deployment errors](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/common-deployment-errors). 

Reviewer 2: ```suggestion
For common deployment errors, refer to [Troubleshoot common Azure deployment errors](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/common-deployment-errors). 
```

1. In the left pane of Azure Portal, Click **Eventhub**.
1. Search for `“SUMOBREventHubNamespace”`. You should find the `“SUMOBREventHubNamespace\<random-string>”` Event Hub Namespace and click it. 
1. Click the **Messages** link.
1. Message summary information appears below the chart. Check that the **Incoming Messages** count is greater than zero.<br/><img src={useBaseUrl('/img/send-data/eventhub-messages-metrics.png')}  style={{border: '1px solid gray'}} alt="eventhub-messages-metrics" width="800"/>

## Verify Service Bus Queue is receiving tasks

> Go to Service Bus Service from the Azure portal and click on `SUMOBRTaskQueueNamespace\<unique string>` Service Bus Namespace. Check that the incoming messages count is greater than zero. <br/><img src={useBaseUrl('/img/send-data/service-bus-metrics.png')}  style={{border: '1px solid gray'}} alt="service-bus-metrics" width="800"/>

Reviewer: ```suggestion
Go to Service Bus Service from the Azure portal and click the `SUMOBRTaskQueueNamespace\<unique string>` Service Bus Namespace. Check that the incoming messages count is greater than zero. <br/><img src={useBaseUrl('/img/send-data/service-bus-metrics.png')}  style={{border: '1px solid gray'}} alt="service-bus-metrics" width="800"/>
```

### Verify Azure Function is not getting Failed

1. Go to the Function App.
1. Navigate to **Application Insights**.<br/><img src={useBaseUrl('/img/send-data/app-insights.png')}  style={{border: '1px solid gray'}} alt="app-insights" width="800"/>
1. Go to **Failures** section.<br/><img src={useBaseUrl('/img/send-data/failures.png')}  style={{border: '1px solid gray'}} alt="failures" width="800"/>
1. Click on the Function Name under Operation Name.

> 1. In the top 3 exception types, click on the count it will open a sample exception.

Reviewer 2: ```suggestion
1. In the top 3 exception types, when you click on the count it will open a sample exception.
```

> 1. Click on any exception it will open an end to end transaction details page where you can click on View all telemetry to view all the logs for that execution.<br/><img src={useBaseUrl('/img/send-data/end-transaction.png')} alt="end-transaction" width="800"/>

Reviewer 2: ```suggestion
1. Click on any exception. It will open an end-to-end transaction details page where you can click on **View all telemetry** to view all the logs for that execution.<br/><img src={useBaseUrl('/img/send-data/end-transaction.png')} alt="end-transaction" width="800"/>
```

### Common Azure function errors

```
ExitCode C0000005
ExitCodeString NATIVE ACCESS VIOLATION
Managed Exception = System.AccessViolationException:Attempted to read or write protected memory. This is often an indication that other memory is corrupt
CallStack - Managed Exception
```

> The above error occurs in certain situations the runtime initiates a host shutdown using `HostingEnvironment.InitiateShutdown`, for example, when an unhandled global exception occurs, when a function TimeoutException is thrown, or when performance counter thresholds are exceeded (HostHealthMonitor).

Reviewer 2: ```suggestion
The above error occurs in certain situations when the runtime initiates a host shutdown using `HostingEnvironment.InitiateShutdown`, for example, when an unhandled global exception occurs, when a function TimeoutException is thrown, or when performance counter thresholds are exceeded (HostHealthMonitor).
```

If you're using this function for quite some time then we recommend redeploying the solution with new ARM templates.

> If the error still persists in BlobTaskProducer function and failure rate > 1% then:

Reviewer: ```suggestion
If the error still persists in the BlobTaskProducer function, and the failure rate > 1%, then:
```

* Increase the timeout of the BlobTaskProducer Function to 30 min in host.json by clicking on **App files**. <br/><img src={useBaseUrl('/img/send-data/azure-error1.png')} alt="error" width="800"/>

* Increase the minimum **Instance count** in app service plan of the `BlobTaskProducer` function by doing the following:

    1. Go to **Monitor > Autoscale**.
    1. Select the resource group in which you deployed the ARM template and select app service plan `(SUMOBRProducerPlan\<suffix\>)` in resource type.<br/><img src={useBaseUrl('/img/send-data/arm-template-suffix.png')} alt="arm-template-suffix" width="800"/>
    1. Click on Manual scale and set min instance count to 2. You can also use auto scaling to save on costs.<br/><img src={useBaseUrl('/img/send-data/azure-manual-scale.png')} alt="azure-manual-scale" width="800"/>

>     If the error still persists in BlobTaskConsumer function and failure rate > 1% then you can migrate from Consumption plan to Premium plan by making changes in the ARM template 

Reviewer: ```suggestion
    If the error still persists in the BlobTaskConsumer function, and the failure rate > 1%, then you can migrate from Consumption plan to Premium plan by making the following changes in the ARM template:
```

### Common Azure function errors

ExitCode C0000005

ExitCodeString NATIVE ACCESS VIOLATION

Managed Exception = System.AccessViolationException:Attempted to read or write protected memory. This is often an indication that other memory is corrupt.

> CallStack - Managed Exception

Reviewer: Why were the tick marks removed from this error message formatting?

Author: It was not working with the ticks present. Now I have modified it accordingly by adding three ticks.

The above error occurs in certain situations the runtime initiates a host shutdown using HostingEnvironment.InitiateShutdown, for example, when an unhandled global exception occurs, when a function TimeoutException is thrown, or when performance counter thresholds are exceeded (HostHealthMonitor).

If you're using this function for quite some time then we recommend redeploying the solution with new ARM templates.

> If the error still persists in BlobTaskProducer function and failure rate > 1% then

Reviewer: ```suggestion
If the error still persists in the BlobTaskProducer function and the failure rate is less than 1%, then
```

> \> Increase the time out of the BlobTaskProducer Function to 30 min in host.json by clicking on Appfiles 

Reviewer: ```suggestion
\> Increase the timeout of the BlobTaskProducer Function to 30 min in host.json by clicking on Appfiles 
```

![error](/img/send-data/azure-error1.png)

> \> Increase the number of min instances in app service plan of the BlobTaskProducer function

Reviewer: ```suggestion
* Increase the minimum **Instance count** in app service plan of the `BlobTaskProducer` function by doing the following:
```

1. Go to Monitor > Autoscale

1. Select the resource group in which you deployed the ARM template and select app service plan `(SUMOBRProducerPlan\<suffix\>)` in resource type

    ![arm-template-suffix](/img/send-data/arm-template-suffix.png)

1. Click on Manual scale and set min instance count to 2. You can also use auto scaling to save on costs.

    ![azure-manual-scale](/img/send-data/azure-manual-scale.png)

If the error still persists in BlobTaskConsumer function and failure rate > 1% then you can migrate from Consumption plan to Premium plan by making changes in the ARM template 

> ```

Reviewer: needs to be formatted
